src/system_database.ts

```diff
@@ -417,7 +426,7 @@ async function insertWorkflowStatus(
             THEN EXCLUDED.executor_id
             ELSE workflow_status.executor_id
           END
-        RETURNING recovery_attempts, status, name, class_name, config_name, queue_name, workflow_deadline_epoch_ms`,
+        RETURNING recovery_attempts, status, name, class_name, config_name, queue_name, workflow_deadline_epoch_ms, executor_id, (xmax = 0) AS inserted`,
```

Comment:
Why does `xmax=0` necessarily mean a fresh insert (as opposed to an upsert, I assume)